src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -60,23 +60,36 @@ namespace winrt::TerminalApp::implementation
         _tabView = _tabRow.TabView();
         _rearranging = false;
 
-        _tabView.TabDragStarting([this](auto&& /*o*/, auto&& /*a*/) {
-            _rearranging = true;
-            _rearrangeFrom = std::nullopt;
-            _rearrangeTo = std::nullopt;
-        });
+        // weak_ptr to this TerminalPage object lambda capturing
+        auto weakThis{ get_weak() };
 
-        _tabView.TabDragCompleted([this](auto&& /*o*/, auto&& /*a*/) {
-            if (_rearrangeFrom.has_value() && _rearrangeTo.has_value() && _rearrangeTo != _rearrangeFrom)
+        _tabView.TabDragStarting([weakThis](auto&& /*o*/, auto&& /*a*/) {
+            if (auto page{ weakThis.get() })
             {
-                auto tab = _tabs.at(_rearrangeFrom.value());
-                _tabs.erase(_tabs.begin() + _rearrangeFrom.value());
-                _tabs.insert(_tabs.begin() + _rearrangeTo.value(), tab);
+                page->_rearranging = true;
+                page->_rearrangeFrom = std::nullopt;
+                page->_rearrangeTo = std::nullopt;
             }
+        });
+
+        _tabView.TabDragCompleted([weakThis](auto&& /*o*/, auto&& /*a*/) {
+            if (auto page{ weakThis.get() })
+            {
+                auto& from{ page->_rearrangeFrom };
+                auto& to{ page->_rearrangeTo };
 
-            _rearranging = false;
-            _rearrangeFrom = std::nullopt;
-            _rearrangeTo = std::nullopt;
+                if (from.has_value() && to.has_value() && to != from)
+                {
+                    auto& tabs{ page->_tabs };
+                    auto tab = tabs.at(to.value());
```

Comment:
This `tab` was `_rearrangeFrom` originally - is this intentional?

Reply:
Nope, good catch! Late nite coding artifact, sorry about that.

Reply:
Nope, good catch! Late nite coding artifact, sorry about that.

Reply:
We've all been there 😝